flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/operators/join/stream/keyselector/AttributeBasedJoinKeyExtractor.java

```diff
@@ -417,14 +417,19 @@ private boolean isCommonConceptualAttributeSet(Set<AttributeRef> eqSet) {
 
     private void processCommonAttributes(List<Set<AttributeRef>> commonConceptualAttributeSets) {
         for (int currentInputId = 0; currentInputId < inputTypes.size(); currentInputId++) {
-            List<AttributeRef> commonAttrsForThisInput =
+            final List<AttributeRef> commonAttrsForThisInput =
                     findCommonAttributesForInput(currentInputId, commonConceptualAttributeSets);
 
             if (commonAttrsForThisInput.isEmpty()) {
+                // This indicates that there is no common join key among all inputs.
+                // In this case, we cannot use a multi-join, so throw an exception.
                 throw new IllegalStateException(
                         "No common attributes found for inputId "
                                 + currentInputId
-                                + ". This indicates a misconfiguration in joinAttributeMap.");
+                                + ". This indicates a misconfiguration in joinAttributeMap,"
```

Comment:
Make the error more user friendly. User doesn't care about internals like joinAttributeMap.